### ISSUE TITLE
custom block with es6 content

### ISSUE THREAD

**Issue by dunkfordyce** [vuejs/vue-loader#1064]
### Version
13.5.0

### Reproduction link
[https://github.com/dunkfordyce/custom-block-es6](https://github.com/dunkfordyce/custom-block-es6)

### Steps to reproduce
Run the sample repo as usual. The custom block `foo` has a loader build/foo-loader.js. The code exports a default function to modify the Component. Because the code is ES6 the function is never run as the loader does not check for an ES6 module, it sees an object instead of a function. 

### What is expected?
The default exported function should be run.

### What is actually happening?
The vue-loader code sees an object instead of a function.

**Comment by LinusBorg** [comment]
I'm not sure if I misunderstand something, but the foo loader doesn'T actually do anything:
```javascript
module.exports = function (source, map) {
    this.callback(null, source, map);
};
```

Did you push the wrong branch or something? Or is my lack of knowledge about  loader internals showing?

**Comment by Jinjiang** [comment]
@LinusBorg I guess I get the point. The `<foo>export default function () { ... }</foo>` cannot be recognized as a function in runtime but an object with a `default` function property. So if we updated the generated code:

```js
if (typeof customBlock === "function") {  customBlock(Component)}
```

That maybe work.

I would create a PR soon if you don't mind.

Thanks.

**Comment by yyx990803** [comment]
closed via #1081
